CONTRIBUTING.md

```diff
@@ -0,0 +1,263 @@
+    * If interested, the configuration for `pre-commit` can be found in `.pre-commit-config.yaml`
+
+*   Finally, we should commit our changes, push them up to our fork and create a pull request.
+    ```bash
+    # Add your changes and make a commit
+    git add {changed files}
+    git commit -m "Something as meaningful as possible"
+    
+    # This will push my_new_branch to your fork located at `origin`
+    git push --set-upstream origin my_new_branch
+    ```
+
+*   At this point, we need to create a PR request to the [automl/autosklearn](https://github.com/automl/auto-sklearn) repository with our new changes.
+    This can be done simply by going to your own forked repo, clicking **'Contribute'**, and selecting
+    the **development** branch of `automl/auto-sklearn`.
+    *   `automl/auto-sklearn` | `development` <- `your-username/auto-sklearn` | `my_new_branch`
+    The reason we don't want to directly merge new PR's into master is to make
+    sure we always have a stable version. With a development branch, we can safely
+    accumulate certain changes and makes sure they all work together before creating
+    a new master version.
+
+*   Now you've got to describe what you've changed.
+    Some key things to include here are:
+    *   A high level overview of what you've done such as fixing a problem or
+        introducing a new feature.
+        If it's a simple doc fix, don't worry too much about this.
+    *   Have you introduced any breaking changes to the API?
+        We may not realise it while reviewing the changes but if you are aware,
+        it definitely helps to tell us!
+    *   Do you think this might have any implications in the future or how would
+        further work on this look like?
+    *   If you've introduced some new feature, write about who might use it,
+        give a breif code sample to show it and let us know how you tested it!
+    *   If you've fixed a bug, write about why this bug exists in the first place,
+        how you solved it and how a test makes sure it won't pop up again!
+
+*   Once you've submitted a PR, we'll probably enable some automatic tests to run.
+    This will make sure all the tests run smoothly, make sure the documentation builds correctly and do some quick check on code quality.
+    You'll be able to see these run in the **Checks** tab or at the bottom of the PR.
+    If you see a red x, that means somethings probably gone wrong which should have been caught by running the tests locally but we also do some checks in environments you were not developing on.
+    Sometimes it is the case that there is a bug unrelated to your changes that cause the tests to fail but we are aware of these.
+    If you see one of these failures, feel free to ask!
+
+*   Meanwhile, we'll also review your code.
+    In the top right you'll see a little **Suggested Reviewers** button so feel free to request a review with that!
```

Comment:
I don't think anyone outside of our organization can see that button. You can check with any other repo.